http_rest_methods/methods.go

```diff
@@ -0,0 +1,65 @@
+package http_rest_methods
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/akitasoftware/go-utils/sets"
+	"github.com/pkg/errors"
+)
+
+type HTTPMethod string
```

Comment:
I am really annoyed that we have to do this ourselves, the constants in net/http are just strings.

Reply:
Go: opinionated in all the wrong places! 🙃